diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7245f52..25af8e5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,8 +23,9 @@ else()
   add_compile_options(-D_GNU_SOURCE)
 endif()
 
-find_package(libvsync 4.2 CONFIG REQUIRED)
-add_library(vsync INTERFACE IMPORTED)
+# find_package(libvsync 4.2 CONFIG REQUIRED) add_library(vsync INTERFACE
+# IMPORTED)
+add_subdirectory(../libvsync libvsync)
 link_libraries(vsync)
 
 include_directories(include)
@@ -32,7 +33,10 @@ include_directories(include)
 add_subdirectory(lib)
 add_subdirectory(mod)
 add_subdirectory(tsano)
-add_subdirectory(test)
+
+if(off)
+  add_subdirectory(test)
+endif()
 
 # Installation
 
