diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5e70f6c..17626f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,11 +43,11 @@ add_subdirectory(deps)
 add_subdirectory(include)
 add_subdirectory(src)
 
-if(PROJECT_IS_TOP_LEVEL)
-  enable_testing()
-  add_subdirectory(test)
-  add_subdirectory(bench/micro)
-  add_subdirectory(bench/lib)
-endif()
+#if(PROJECT_IS_TOP_LEVEL)
+#  enable_testing()
+#  add_subdirectory(test)
+#  add_subdirectory(bench/micro)
+#  add_subdirectory(bench/lib)
+#endif()
 
 install(TARGETS dice dice.h EXPORT ${PROJECT_TARGETS})
diff --git a/src/dice/CMakeLists.txt b/src/dice/CMakeLists.txt
index f956c31..5fd40ec 100644
--- a/src/dice/CMakeLists.txt
+++ b/src/dice/CMakeLists.txt
@@ -80,6 +80,7 @@ foreach(SRC ${DISPATCH_SRCS})
 endforeach()
 
 # dice libraries
+set(SRCS)
 set(SRCS mempool.c rbtree.c loader.c pubsub.c ${SRCS})
 add_library(dice.o OBJECT ${SRCS})
 target_link_libraries(dice.o PUBLIC dice.h)
