diff --git a/CMakeLists.txt b/CMakeLists.txt
index 09f9302..cb5c316 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -38,7 +38,7 @@ if(VSYNC_MEMCPY_S_EXISTS)
     target_compile_definitions(vsync INTERFACE "VSYNC_MEMCPY_S_EXISTS")
 endif()
 
-if(CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
+if(off)
     # ##########################################################################
     # Important Config must be set
     # ##########################################################################
